Formula/d/duplicity.rb

```diff
@@ -397,8 +397,8 @@ class Duplicity < Formula
   end
 
   resource "pyopenssl" do
-    url "https://files.pythonhosted.org/packages/54/9a/2a43c5dbf4507f86f7c43cba4195d5e25a81c988fd7b0ea779dfc9c6973f/pyOpenSSL-21.0.0.tar.gz"
-    sha256 "5e2d8c5e46d0d865ae933bef5230090bdaf5506281e9eec60fa250ee80600cb3"
+    url "https://files.pythonhosted.org/packages/c1/d4/1067b82c4fc674d6f6e9e8d26b3dff978da46d351ca3bac171544693e085/pyopenssl-24.3.0.tar.gz"
```

Comment:
This will get auto-reverted on next version bump due to dependency resolver, e.g. prior attempt (https://github.com/Homebrew/homebrew-core/pull/182946) got accidentally reverted from running `brew update-python-resources`

Also, this specific minor version doesn't look compatible with `pydrive2` dependency:
* https://github.com/iterative/PyDrive2/blob/main/pyproject.toml#L30
* https://github.com/iterative/PyDrive2/issues/361

---

Probably need to update to newer version as 3.0.2 has a problematic pin https://gitlab.com/duplicity/duplicity/-/blob/rel.3.0.2/requirements.txt?ref_type=tags#L5
```requirements
cryptography==3.4.8
```

This is why 21.0.0 is picked rather than 24.2.1

Reply:
Ah. I see.
Starting from `duplicity 3.0.3`, cryptography is no longer a requirement. 
https://gitlab.com/duplicity/duplicity/-/blob/rel.3.0.3/requirements.txt?ref_type=tags#L5

Latest version is 3.0.3.2. Make sense to skip this and go straight to release that version.

Reply:
Looks like autobump may not have triggered after 3.13 migration (https://github.com/Homebrew/homebrew-core/pull/194416) as there is no release with https://gitlab.com/duplicity/duplicity/-/commit/98e7113bdb20f76b4d636a996a63afb3bda809cf so resolver skips recent versions.

```console
❯ /opt/homebrew/opt/python@3.13/libexec/bin/python -m pip install -q --disable-pip-version-check --dry-run --ignore-installed --report=/dev/stdout 'duplicity==3.0.3.2'
ERROR: Ignored the following yanked versions: 0.8.21.post1, 0.8.21.post2.dev4, 0.8.21.post2, 0.8.21.post6, 0.8.21.post7, 0.8.22.dev4, 1.2.0, 2.0.0, 2.2.3.dev18
ERROR: Ignored the following versions that require a different python version: 3.0.1 Requires-Python <3.13,>=3.8; 3.0.2 Requires-Python <3.13,>=3.8; 3.0.3 Requires-Python <3.13,>=3.8; 3.0.3.1 Requires-Python <3.13,>=3.8; 3.0.3.2 Requires-Python <3.13,>=3.8
```